### ISSUE TITLE
Airflow Rates for Fixed Size Equipment

### ISSUE THREAD

**Issue by jmaguire1** [NREL/OpenStudio-BEopt#107]
When using fixed equipment sizes, we should be trying to use the airflow rates calculated in sizing .py whenever possible. In a lot of cases, sizing.py just calculates the airflow rate as x cfm/ton of cfm/(kBtu/h), which we can implement if a fixed size is used. I'll add a list to this issue of how we do the sizing calculations for each piece of HVAC equipment and work with Joe to use those calculations where possible in our measures. This should also give us a way to better compare HVAC fan/pump energy between BEopt and OpenStudio to see if sizing is the primary reason for the discrepancy.

**Comment by shorowit** [comment]
We could consider exposing the airflow rate (cfm? cfm/ton?) as a measure argument as well. We could then require the user either pick Autosize for both capacity and airflow rate or a number for both.

**Comment by joseph-robertson** [comment]
Residential central air conditioner measure:
```ruby
supply.CFM_TON_Rated = HVAC.calc_cfm_ton_rated(acRatedAirFlowRate, acFanspeedRatio, acCapacityRatio)
clg_coil.setRatedAirFlowRate(supply.CFM_TON_Rated[0] * acOutputCapacity * OpenStudio::convert(1.0,"Btu/h","ton").get * OpenStudio::convert(1.0,"cfm","m^3/s").get)
```

**Comment by jmaguire1** [comment]
From sizing.py, here's how we do the calculations for furnaces:

`unit.supply.Heat_AirFlowRate = heating load / (1.1 * sim.site.acf * (unit.supply.htg_supply_air_temp - mj8.heating_setpoint))`

Some terminology here: 

- heating load is capacity of the furnace
- sim.site.acf is an altitude correction factor for the site (to account for variations in air density with elevation). Pretty sure we're not currently calculating that anywhere, but it would be:

`#Calculate the altitude correction factor (ACF) for the site
mj8.acfs = [1.0, 0.97, 0.93, 0.89, 0.87, 0.84, 0.80, 0.77, 0.75, 0.72, 0.69, 0.66, 0.63]
alt_cnt = int(weather.header.Altitude / 1000.0)
sim.site.acf = MathTools.interp2(weather.header.Altitude, alt_cnt * 1000, (alt_cnt + 1) * 1000, mj8.acfs[alt_cnt], mj8.acfs[alt_cnt + 1])`

- unit.supply.htg_supply_air_temp is the user entered supply air temperature.
- mj8.heating_setpoint is the heating setpoint used in Manual J (70 F)

I think we do a very similar calculation for boilers, but I'll go through sizing.py to confirm. For air conditioners, MSHPs and ASHPs, it looks like the fan size (cfm/ton) is already input, so there's not going to be anything special to do for fixed size equipment.

**Comment by shorowit** [comment]
We should append to the airflow rate argument descriptions that the value only applies if using fixed capacity.

**Comment by shorowit** [comment]
Based on a conversation with @jonwinkler, we have the following work-in-progress proposal for the AC, ASHP, and MSHP measures.
- We should remove the Rated Airflow Rate inputs. 
- The measures should have inputs for Capacity, Airflow Rate (actual, not rated), and SHR:
  - If Capacity is autosized, Airflow Rate and SHR inputs are ignored and we use autosize. Rated Airflow Rate is autosized.
  - If Capacity is user-specified, Airflow Rate and SHR entered values are used. Rated Airflow Rate is calculated using these cfm/ton values:
     - 386.1 for 1-speed AC
     - 355.2 for 2-speed AC
     - 315.8 for variable-speed AC
     - 394.2 for 1-speed ASHP
     - 344.1 for 2-speed ASHP
     - 315.8 for variable-speed AC
     - (Not sure for MSHP ???)

cc @ejhw

**Comment by shorowit** [comment]
This should all get resolved with our HVAC autosizing measure.

**Comment by shorowit** [comment]
Partially implemented in https://github.com/NREL/OpenStudio-BEopt/commit/6be7a6893e7fcfc9cd2b2577adba955f74a65645. The rest will be implemented in #8.
